doc/crypto/api/ops/ka.rst

```diff
@@ -114,6 +122,91 @@ Key agreement algorithms
 Standalone key agreement
 ------------------------
 
+.. function:: psa_key_agreement
+
+    .. summary::
+        Perform a key agreement and return the shared secret as a derivation key.
+
+    .. param:: psa_algorithm_t alg
```

Comment:
Another parameter ordering consistency issue.

This API follows [`psa_raw_key_agreement()`](https://arm-software.github.io/psa-api/crypto/1.1/api/ops/ka.html#c.psa_raw_key_agreement), placing the algorithm before the key. Which happens to be the odd one out - _every_ other function that takes a key and an algorithm in this API, has the key parameter before the algorithm.

Should we make this API fit with the general pattern? - If so, is there any preference for _both_ keys coming before the algorithm ("Using these keys, compute key agreement `alg`"):

```
psa_status_t psa_key_agreement(psa_key_id_t private_key,
                               const uint8_t * peer_key,
                               size_t peer_key_length,
                               psa_algorithm_t alg,
                               const psa_key_attributes_t * attributes,
                               psa_key_id_t * key);
```

... or just the the `psa_key_id_t` parameter ("Using my key, compute key agreement `alg` with their key"):

```
psa_status_t psa_key_agreement(psa_key_id_t private_key,
                               psa_algorithm_t alg,
                               const uint8_t * peer_key,
                               size_t peer_key_length,
                               const psa_key_attributes_t * attributes,
                               psa_key_id_t * key);
```

... ?

Reply:
I suspect that the order of parameters of `psa_raw_key_agreement` comes from `psa_key_derivation_key_agreement`, which was written first. There the order is: operation, step, keys. The algorithm is extracted from the operation, so `psa_raw_key_agreement` has the algorithm first.

But I agree that it would have been better to put the algorithm in the same place as other functions that take an input key and an algorithm.

Either way, the input keys should be kept together. There's no reason to put the algorithm between `private_key` and `peer_key`.

Reply:
So that narrows it to two options: parameter consistency with `psa_raw_key_agreement()` or with 'all the other APIs'? - do I detect a preference for the latter? (That is my preference and leave raw key agreement as the odd one out)

Reply:
Yes, I prefer to leave `psa_raw_key_agreement` as the odd one out.

Reply:
I've reordered the parameters: see https://github.com/ARM-software/psa-api/pull/101/commits/18c73a6a8fefda442d5c20a28ae9752cd75827c0